addons/account/report/account_invoice_report.py

```diff
@@ -95,12 +95,12 @@ def _select(self):
                 move.invoice_date_due,
                 uom_template.id                                             AS product_uom_id,
                 template.categ_id                                           AS product_categ_id,
-                line.quantity / NULLIF(COALESCE(uom_line.factor, 1) / COALESCE(uom_template.factor, 1), 0.0) * (CASE WHEN move.move_type IN ('in_invoice','out_refund','in_receipt') THEN -1 ELSE 1 END)
+                line.quantity / NULLIF(COALESCE(uom_line.factor, 1) * COALESCE(uom_template.factor, 1), 0.0) * (CASE WHEN move.move_type IN ('in_invoice','out_refund','in_receipt') THEN -1 ELSE 1 END)
                                                                             AS quantity,
                 -line.balance * currency_table.rate                         AS price_subtotal,
                 -COALESCE(line.balance
                    / NULLIF(line.quantity, 0.0)
-                   / NULLIF(COALESCE(uom_line.factor, 1), 0.0)
+                   * NULLIF(COALESCE(uom_line.factor, 1), 0.0)
```

Comment:
What we had before was 
`-line.balance / NULLIF(COALESCE(uom_line.factor, 1) / COALESCE(uom_template.factor, 1), 0.0)`
or more simply `-balance / (uom_factor / template_factor)`
= `-balance / uom_factor * template_factor`

So the `*` sign should be for the line below if I trust the version that was there before the bugfix mentionned.

Reply:
I made a test just to be sure of the math

Suppose to have a Product with unit of measure KG and 2 invoice line as follows:

```
  500g  @    1$/g   ->  500$
    1kg @ 1000$/kg  -> 5000$
```

before my patch we used as AveragePrice the formula `balance / (uom_factor / template_factor)`
The report show

|Total|Quantity|AveragePrice|
|------|------------|------------------|
|5,500.00|5.50|2,500.25|

so it seems it was not correct from the beginning
after my patch we have the formula `balance / quantity / (uom_factor / template_factor)`
|Total|Quantity|AveragePrice|
|------|------------|------------------|
|5,500.00|5.50|500.00|

By using the current proposal, we would have the formula 

`-balance / quantity * (uom_factor / template_factor) `

resulting in

|Total|Quantity|AveragePrice|
|------|------------|------------------|
|5,500.00|5.50|1000.00|

which imho **looks correct**

Reply:
Yeah, don't use multiple division signs on the "same line". The order of execution is not well defined and may vary.
Use a notation like this instead `(a * b * c) / (d * e * f)`, which is a lot easier to understand.